test/BinTestCases.test.js

```diff
@@ -41,115 +41,114 @@ function convertToArrayOfLines(outputArray) {
 const defaultArgs = loadOptsFile(path.join(casesPath, "test.opts"));
 
 describe("BinTestCases", function() {
-	const categoryDirectories = fs.readdirSync(casesPath).filter((folder) => {
-		return fs.statSync(path.join(casesPath, folder)).isDirectory();
-	});
+	const tests = findTestsRecursive(casesPath);
+
+	tests.forEach(testDirectory => {
+		const testName = testDirectory.replace(casesPath, "");
+		const testArgs = getTestSpecificArguments(testDirectory) || defaultArgs;
+		const testAssertions = require(path.join(testDirectory, "test.js"));
+		const outputPath = path.join(path.resolve(casesPath, "../js/bin"), testName);
+
+		const cmd = `${path.resolve(process.cwd(), "bin/webpack.js")}`;
+		const args = testArgs.concat(["--output-path", `${outputPath}`]);
+		const opts = {
+			cwd: path.resolve("./", testDirectory)
+		};
+
+		const asyncExists = fs.existsSync(path.join(testDirectory, "async"));
 
-	const categories = categoryDirectories.map(function(categoryDirectory) {
-		return {
-			name: categoryDirectory,
-			tests: fs.readdirSync(path.join(casesPath, categoryDirectory))
+		const env = {
+			stdout: [],
+			stderr: [],
+			error: []
 		};
-	});
 
-	categories.forEach(function(category) {
-		describe(category.name, function() {
-
-			category.tests.forEach(function(testName) {
-				const testDirectory = path.join(casesPath, category.name, testName);
-				const testArgs = getTestSpecificArguments(testDirectory) || defaultArgs;
-				const testAssertions = require(path.join(testDirectory, "test.js"));
-				const outputPath = path.join(path.resolve(casesPath, "../js/bin"), category.name, testName);
-
-				const cmd = `${path.resolve(process.cwd(), "bin/webpack.js")}`;
-				const args = testArgs.concat(["--output-path", `${outputPath}`]);
-				const opts = {
-					cwd: path.resolve("./", testDirectory)
-				};
-
-				const asyncExists = fs.existsSync(path.join(testDirectory, "async"));
-
-				const env = {
-					stdout: [],
-					stderr: [],
-					error: []
-				};
-
-				if(asyncExists) {
-					describe(testName, function() {
-						it("should run successfully", function(done) {
-							this.timeout(10000);
-							const child = spawn([cmd].concat(args), opts);
-
-							child.on("close", function(code) {
-								env.code = code;
-							});
-
-							child.on("error", function(error) {
-								env.error.push(error);
-							});
-
-							child.stdout.on("data", (data) => {
-								env.stdout.push(data);
-							});
-
-							child.stderr.on("data", (data) => {
-								env.stderr.push(data);
-							});
-
-							setTimeout(() => {
-								if(env.code) {
-									done(`Watch didn't run ${env.error}`);
-								}
-
-								const stdout = convertToArrayOfLines(env.stdout);
-								const stderr = convertToArrayOfLines(env.stderr);
-								testAssertions(stdout, stderr, done);
-								child.kill();
-							}, 8000); // wait a little to get an output
-						});
+		if(asyncExists) {
+			describe(testName, function() {
```

Comment:
I think all `function()` could be refactored to an arrow, but that's entirely additional chore so feel free to ignore 😛